llvm/lib/Transforms/Instrumentation/AddressSanitizer.cpp

```diff
@@ -1240,6 +1269,116 @@ struct FunctionStackPoisoner : public InstVisitor<FunctionStackPoisoner> {
 
 } // end anonymous namespace
 
+// Append a new argument "launch_data" to user's spir_kernels
+static void ExtendSpirKernelArgs(Module &M, FunctionAnalysisManager &FAM) {
+  SmallVector<Function *> SpirFixupFuncs;
+  for (Function &F : M) {
+    if (F.getCallingConv() == CallingConv::SPIR_KERNEL) {
+      SpirFixupFuncs.emplace_back(&F);
+    }
+  }
+
+  SmallVector<std::pair<Function *, Function *>> SpirFuncs;
+  const int LongSize = M.getDataLayout().getPointerSizeInBits();
+  auto *IntptrTy = Type::getIntNTy(M.getContext(), LongSize);
```

Comment:
It seems like there is a helper for that: [`DataLayout::getIntPtrType`](https://llvm.org/doxygen/classllvm_1_1DataLayout.html#a7a2619d0e489a4ba9c19a0d86a041d59)

Reply:
done